src/main/java/com/google/sps/servlets/DeleteTutorSessionServlet.java

```diff
@@ -44,12 +55,92 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         String studentID = Optional.ofNullable((String)request.getSession(false).getAttribute("userId")).orElse("-1");
         long id = Long.parseLong(request.getParameter("id"));
 
+        TutorSession tutorSession = datastore.getScheduledSession(id);
+
+        System.out.println(tutorSession.getStudentID());
+
+        Entity studentEntity = datastore.getStudentForUserId(tutorSession.getStudentID());
+        Entity tutorEntity = datastore.getTutorForUserId(tutorSession.getTutorID());
+        TimeRange timeslot = tutorSession.getTimeslot();
+
+        boolean testTutorEmail = sendCancellationEmailToTutor(studentEntity, tutorEntity, timeslot);
+        boolean testStudentEmail = sendCancellationEmailToStudent(studentEntity, tutorEntity, timeslot);
+
         // Remove tutor session
         datastore.deleteTutorSession(studentID, id);
 
         String json = new Gson().toJson(datastore.getScheduledSession(id));
         response.setContentType("application/json;");
-        response.getWriter().println(json);
+        response.getWriter().println(json + "testTutorEmail: " + testTutorEmail + " testStudentEmail: " + testStudentEmail);
         return;
     }
+
+    public boolean sendCancellationEmailToTutor(Entity studentEntity, Entity tutorEntity, TimeRange timeslot) {
```

Comment:
If the only different between `sendCancellationEmailToTutor` and `sendCancellationEmailToStudent` is the message, maybe we can make `message` a parameter and use a single method. I think we only use the studentEntity in the tutor method for the student's name and the tutorEntity in the student method for the tutor's name. We could also do all of the date stuff outside of this method to create the message.

So we could have the method be `sendCancellationEmail(Entity entity, String message)`. Does that make sense?

Reply:
Yes, this makes sense. I'll implement this for the cancellation and scheduling emails.